stores/firmwareStore.ts

```diff
@@ -178,7 +179,18 @@ export const useFirmwareStore = defineStore('firmware', {
       };
       const transport = new Transport(this.port, true);
       const espLoader = await this.connectEsp32(transport, terminal);
-      const appContent = await this.fetchBinaryContent(fileName);
+      let appContent = await this.fetchBinaryContent(fileName, true);
+      const tz = Intl.DateTimeFormat().resolvedOptions().timeZone;
+      // get posix timezone string based on browser locale
+      const posixTz = timezones[tz as keyof typeof timezones] || 'CST6CDT,M3.2.0,M11.1.0';
+      appContent.replace('tzplaceholder                                         ', posixTz);
```

Comment:
The number of characters replaced has to be the same as the number of characters in your tz string. You may need to null tereminate, too.